### PR TITLE
feat: BannerView

### PR DESCRIPTION
**Pull request by fcs-ts**
replaces #15

### CODE REVIEW

File: TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class BannerView(
+    context: Context,
+    config: BannerConfig,
+    path: String,
+    location: String,
+    clickCallback: (String, EntityType) -> Unit
+) : ImageView(context) {
+    init {
+        val result = runBannerAuction(config)
```
Comment (PabloReszczynski) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt:42]: can this function throw? I'm guessing the `null` value is when there are no winners, right?

File: TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import com.topsort.analytics.Analytics
+import com.topsort.analytics.model.Placement
+import com.topsort.analytics.model.auctions.EntityType
+
+class BannerView(
+    context: Context,
+    config: BannerConfig,
+    path: String,
+    location: String,
+    clickCallback: (String, EntityType) -> Unit
```
Comment (PabloReszczynski) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt]: Is this the callback needed for navigation after clicking?

File: TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package com.topsort.analytics.banners
+
+import android.content.Context
+import android.widget.ImageView
+import coil.load
+import com.topsort.analytics.Analytics
+import com.topsort.analytics.model.Placement
+import com.topsort.analytics.model.auctions.EntityType
+
+class BannerView(
```
Comment (PabloReszczynski) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt:22]: Please add extensive docs on how to properly use this View class.

Comment (PabloReszczynski) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt:22]: It's generally a standard to make library classes `final`

**Review by PabloReszczynski — CHANGES_REQUESTED** [review]

File: TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt (hunk 1 above)
Comment (fcs-ts) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt:42]: Yes, `null` is for no winners. I am not sure if the function can throw downstream, though. @anonvt can we throw on the auction client?

File: TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt (hunk 2 above)
Comment (fcs-ts) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt]: yes, I have to change the signature according to the article you sent me, though

Comment (PabloReszczynski) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt]: I'd change it to `onClick`, or `onTap` in that case

File: TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt (hunk 3 above)
Comment (fcs-ts) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt:22]: kotlin classes are `final` by default, you have to specify the `open` keyword to allow inheritance

File: TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt (hunk 1 above)
Comment (PabloReszczynski) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt:42]: Don't know if kotlin allow putting `throws` like java, to add the error to the type signature of the function

Comment (PabloReszczynski) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerView.kt:42]: imo the client object should throw, but the view should be able to handle errors

**Review by anonvt — APPROVED** [review]
Other than Palbo's comment regarding some docs, LGTM

**Comment by PabloReszczynski** [comment]
> Other than Palbo's comment regarding some docs, LGTM

Palbo

**Comment by fcs-ts** [comment]
> Palbo

Colo-Colo won a match yesterday, so I guess it's fitting

**Comment by PabloReszczynski** [comment]
@fcs-ts @anonvt what is missing for this to be merged?

**Comment by fcs-ts** [comment]
> what is missing for this to be merged?

@PabloReszczynski I had some trouble getting this to actually run auctions. I made some questions to @anonvt on how the new service handler should be setup (that we also want to have in the README/docs).

**Comment by anonvt** [comment]
Hopefully the issues with the ServiceSettings has been by #24 
Otherwise LGTM

**Comment by fcs-ts** [comment]
@PabloReszczynski Networking issues are fixed and I think I covered the docs
@anonvt When testing this I noticed a few issues when parsing the banner auctions responses, I would appreciate some review on these new changes :)

**Review by PabloReszczynski — APPROVED** [review]

**Pull request merged**